### PR TITLE
feat: [DHIS2-17726] Plugins in Profile Widget

### PR DESCRIPTION
**Pull request by eirikhaugstulen**
Summary:
- Allow and render form field plugins in the Profile Widget
- Copied the same logic from Enrollment Factory

### CODE REVIEW

**Comment by eirikhaugstulen** [comment]
Added a test case for developers to set up the plugin: https://dhis2.atlassian.net/browse/DHIS2-17727.

File: src/core_modules/capture-core/components/WidgetProfile/DataEntry/FormFoundation/RenderFoundation.js [hunk 1]
```diff
@@ -97,23 +110,63 @@ const buildElementsForSection = async ({
...
-        const element = await buildDataElement(
-            trackedEntityAttribute,
-            trackedEntityAttributes,
-            optionSets,
-            querySingleResource,
-            minorServerVersion,
-        );
-        element && section.addElement(element);
+        if (isPluginElement(trackedEntityAttribute)) {
+            const pluginElement = ((trackedEntityAttribute: any): PluginElement);
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/WidgetProfile/DataEntry/FormFoundation/RenderFoundation.js:122]: Nice Flow type conversion 👏

**Review by simonadomnisoru — APPROVED** [review]
LGTM 🎉 !

**Review by geethaalwan — APPROVED** [review]
Tested successfully on 2.42,1.41.1,2.40.5,2.39.7 versions

**Pull request merged**